apps/wallet/src/ui/app/pages/dapp-tx-approval/index.tsx

```diff
@@ -208,10 +222,17 @@ export function DappTxApprovalPage() {
     }[] = useMemo(() => {
         switch (txRequest?.tx.type) {
             case 'v2': {
+                const moveCallTxn = txRequest.tx.data
+                    .data as MoveCallTransaction;
                 return [
+                    { label: 'Transaction Type', content: 'MoveCall' },
```

Comment:
V2 transactions aren't just move calls.  They can be any transaction type (Pay, TransferObject, MoveCall, etc). You should adjust this to account for that.

Reply:
reverted to `tx.data.kind`